homework 13

ZubenkoHalyna opened the pull request.

ZubenkoHalyna committed:
implementation added

ZubenkoHalyna committed:
comment deleted

ZubenkoHalyna committed:
Dao classes made singletons with private constructors;
Dao classes marked with annotation @Dao instead of interfaces;

ZubenkoHalyna committed:
Method getDao moved to Injector class

ZubenkoHalyna committed:
Refactoring: Dao annotation check moved to getDaoImpl method

ZubenkoHalyna committed:
Injector.getInstance made static

RomanDubovskyi commented on src/main/java/core/basesyntax/factory/DaoFactory.java:
```diff
@@ -0,0 +1,10 @@
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+
+public interface DaoFactory {
```
цей інтерфейс тут виглядає зайвим давай приберемо його

RomanDubovskyi commented on src/main/java/core/basesyntax/db/Storage.java:
```diff
@@ -0,0 +1,16 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage<T> {
```
цей клас в нас імітує базу данних, в ньому повинні знаходитись лише данні без логіки ( логіку ми реалзовуємо в слої dao)

RomanDubovskyi commented on src/main/java/core/basesyntax/dao/UserDaoImpl.java:
```diff
@@ -0,0 +1,33 @@
+        private static UserDao userDao = new UserDaoImpl();
+    }
+
+    public static UserDao getInstance() {
```
за створення об'єктів у нас відповідальна Factory

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,41 @@
+                field.set(instance, getDaoImpl(field.getType()));
+            }
+        }
+        return instance;
```
а що буде якщо аннотація Dao не буде стояти над класом? давай будемо кидати ексепшин, який самі створимо?

RomanDubovskyi reviewed (commented):
Спробуй відрефакторити структуру і логіку коду згідно з відео Богдана, щоб дотримувати паттернів, які ми юзаємо в цьому завданні

ZubenkoHalyna committed:
deleted logic from Storage;
Custom exception added;
Factory used instead of Singleton;
Factory interface deleted;

ZubenkoHalyna committed:
storage made static

ZubenkoHalyna committed:
refactoring

RomanDubovskyi commented on src/main/java/core/basesyntax/factory/Factory.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Factory {
+    private static class BetDaoHolder {
+        private static BetDao betDao = new BetDaoImpl();
```
```suggestion
        private static BetDao betDao;
```

RomanDubovskyi commented on src/main/java/core/basesyntax/factory/Factory.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    public BetDao getBetDao() {
+        return BetDaoHolder.betDao;
```
```suggestion
         if (betDao == null) {
            betDao = new BetDaoImpl();
        }
        return betDao;
```

RomanDubovskyi commented on src/main/java/core/basesyntax/factory/Factory.java:
```diff
@@ -0,0 +1,24 @@
+import core.basesyntax.dao.UserDaoImpl;
+
+public class Factory {
+    private static class BetDaoHolder {
```
цей клас тут зайвий, може просто у Factory записати дві змінні і методи для отримання їх об'єктів

RomanDubovskyi reviewed (approved):
зміни у класі фекторі свої методи щоб вони працювали як раніше в тебе було(синглтони), решту ок, гарна робота 👍
